src/main/scala/org/constellation/checkpoint/CheckpointAcceptanceService.scala

```diff
@@ -36,6 +45,17 @@ class CheckpointAcceptanceService[F[_]: Concurrent](
 
   val logger: SelfAwareStructuredLogger[F] = Slf4jLogger.getLogger[F]
 
+  def acceptWithNodeCheck(checkpoint: FinishedCheckpoint): F[Unit] =
+    LiftIO[F].liftIO(dao.cluster.getNodeState).flatMap { nodeState =>
+      (nodeState, checkpoint.checkpointCacheData.checkpointBlock) match {
+        case (_, None)                  => Sync[F].raiseError[Unit](MissingCheckpointBlockException)
+        case (NodeState.Ready, Some(_)) => accept(checkpoint)
+        case (NodeState.DownloadCompleteAwaitingFinalSync, Some(_)) =>
+          LiftIO[F].liftIO(dao.snapshotService.syncBufferAccept(checkpoint))
```

Comment:
You have a dependency for `snapshotService` so it should be used instead of `dao.`